h2o-core/src/test/java/water/rapids/ast/prims/assign/AstAppendTest.java

```diff
@@ -0,0 +1,63 @@
+package water.rapids.ast.prims.assign;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.H2O;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+import water.rapids.Rapids;
+import water.rapids.Val;
+import water.rapids.vals.ValFrame;
+import water.util.TwoDimTable;
+
+import static org.junit.Assert.assertEquals;
+
+public class AstAppendTest extends TestUtil {
+
+  @BeforeClass
+  static public void setup() { stall_till_cloudsize(1); }
+
+  private Frame fr = null;
+
+  @Before
+  public void beforeEach() {
+
+  }
+
+  @Test
+  public void AppendColumnTest() {
+
+    fr = new TestFrameBuilder()
+            .withName("testFrame")
+            .withColNames("ColA", "ColB")
+            .withVecTypes(Vec.T_NUM, Vec.T_NUM)
+            .withDataForCol(0, ard(1, 2))
+            .withDataForCol(1, ard(2, 5))
+            .build();
+
+    String tree = "( append testFrame ( / (cols testFrame [0]) (cols testFrame [1])) 'appended' )";
+    Val val = Rapids.exec(tree);
+    if (val instanceof ValFrame)
+      fr = val.getFrame();
+
+    assertEquals(2, fr.numRows());
+
+    TwoDimTable twoDimTable = fr.toTwoDimTable();
+    System.out.println(twoDimTable.toString());
+
+    assertEquals(fr.vec(2).at(0L), 0.5, 1e-6);
+    assertEquals(fr.vec(2).at(1L), 0.4, 1e-6);
+
+  }
+
+  @After
+  public void afterEach() {
+    H2O.STORE.clear();
```

Comment:
What is the purpose? This is a very low-level API.

Reply:
@michalkurka Sorry, I'm not receiving notifications for some reasons. Just searched through my Mail and there are no notifications from here. Even though I'm subscribed. Yes.... I will have to make sure that there is no leaked keys. Since then I have also found Scope() class with which we can control keys(but it is experimental API). I will definitely get back to this topic.

Reply:
@michalkurka found the problem with my notifications. They very filtered and copied to another folder(not Inbox)... this is a legacy from my previous job.